File: September 2024/Alternate positive and negative numbers (14 sept).js
//JAVA CODE 

class Solution {
     void rearrange(ArrayList<Integer> arr) {
        // code here
        ArrayList<Integer> positive = new ArrayList<>();
        ArrayList<Integer> negative = new ArrayList<>();
        for(int x:arr){
            if(x>=0)positive.add(x);
            else negative.add(x);
        }
        int i=0,j=0,k=0;
        while(i<positive.size() && j<negative.size()){
            if(k%2==1)arr.set(k++,negative.get(j++));
            else arr.set(k++,positive.get(i++));
        }
        while(i<positive.size())arr.set(k++,positive.get(i++));
        while(j<negative.size())arr.set(k++,negative.get(j++));
    }
}

File: September 2024/Alternate_positive_and_negative_numbers(14 Sept).cpp
class Solution {
  public:
    void rearrange(vector<int> &arr) {
        
        
        // code here
        if(arr.size() > 1){
        vector<int> pos;
        vector<int> neg;
        
        for(int i = 0; i < arr.size(); i++){
            
            if(arr[i] < 0){
                
                neg.push_back(arr[i]);
            }
            else{
                
                pos.push_back(arr[i]);
            }
        }
        if(pos.empty() && !neg.empty()){
            
            arr = neg;
        }
        else if(neg.empty() && !pos.empty()){
            
            arr = pos;
        }
        else{
        arr.clear();
        int i = 0;
        int j = 0;
        
        while(!pos.empty() && !neg.empty() && i < pos.size() && j < neg.size()){
            
            arr.push_back(pos[i]);
            arr.push_back(neg[j]);
            i++;j++;
        }
        
        if(i <= pos.size()-1){
            
            while(i <= pos.size()-1){
                
                arr.push_back(pos[i]);
                i++;
            }
        }
        else if(j <= neg.size()-1){
            
            while(j <= neg.size()-1){
                
                arr.push_back(neg[j]);
                j++;
            }
        }
        }
        }
        
        
        
    }
};